### PR TITLE
Use OrdinalIgnoreCase to lookup script breakpoints

### LINKED ISSUE #4668: Can't debug when using PowerShell 7.4
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [_open and closed_ issues](https://github.com/PowerShell/vscode-powershell/issues?q=is%3Aissue) to ensure it has not already been reported.
- [X] I have read the [troubleshooting](https://github.com/PowerShell/vscode-powershell/blob/main/docs/troubleshooting.md) guide.
- [X] I am sure this issue is with the _extension itself_ and does not reproduce in a standalone [PowerShell](https://github.com/PowerShell/PowerShell/issues/new/choose) instance.
- [X] I have verified that I am using the latest version of Visual Studio Code and the PowerShell extension.
- [X] If this is a security issue, I have read the [security issue reporting guidance](https://github.com/PowerShell/vscode-powershell/blob/main/SECURITY.md).

### Summary

It seems that VSCode doesn't stop on breakpoints when debugging in PS 7.4

### PowerShell Version

```console
Name                           Value
----                           -----
PSVersion                      7.4.0-preview.4
PSEdition                      Core
GitCommitId                    7.4.0-preview.4
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0.}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visual Studio Code Version

```console
1.80.1
x64
```

### Extension Version

```console
ms-vscode.powershell@2023.6.0
```

### Steps to Reproduce

1. Create empty file
2. Add some code

```

Write-Host "Test 1"

Write-Host "Test 2"
```

3. Set breakpoint

4. Press F5
5. Notice it doesn't stop

### Visuals

Debugging not working

### Logs

_No response_

### LINKED ISSUE #20044: Regression in debugger broke setting breakpoints in VS Code PowerShell extension
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Reproduction steps are in https://github.com/PowerShell/vscode-powershell/issues/4668. The latest preview, `preview.4` breaks setting breakpoints in the PowerShell extension for VS Code. I was able to reproduce this with a build from `master` and then noted it no longer reproduced after reverting the changes introduced by https://github.com/PowerShell/PowerShell/pull/14953.

### Expected behavior

```console
The breakpoints set in the extension should actually be set in PowerShell
```

### Actual behavior

```console
The breakpoints are never set (or maybe hit?)
```

### Error details

_No response_

### Environment data

```powershell
preview.4
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by fflaten**
# PR Summary

Reverts to `OrdinalIgnoreCase` string comparison when looking up breakpoints for a specific script.

## PR Context

Unintended regression introduced in #14953. Pre-PR the option was set [here](https://github.com/nohwnd/PowerShell/blob/ec0eb220626c5d30e8feeb1e9519d028fe67e906/src/System.Management.Automation/engine/debugger/debugger.cs#L1313-L1320).

Replaces #20042
Fix #20044
Fix PowerShell/vscode-powershell#4668

/cc @andyleejordan @nohwnd 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by daxian-dbw — APPROVED** [review]
LGTM. Thanks @fflaten!

File: src/System.Management.Automation/engine/debugger/debugger.cs [hunk 1]
```diff
@@ -982,7 +982,7 @@ internal ScriptDebugger(ExecutionContext context)
             _inBreakpoint = false;
             _idToBreakpoint = new ConcurrentDictionary<int, Breakpoint>();
             // The string key is function context file path. The int key is sequencePoint index.
-            _pendingBreakpoints = new ConcurrentDictionary<string, ConcurrentDictionary<int, LineBreakpoint>>();
+            _pendingBreakpoints = new ConcurrentDictionary<string, ConcurrentDictionary<int, LineBreakpoint>>(StringComparer.OrdinalIgnoreCase);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/debugger/debugger.cs:985]: Should it be only for Windows?

Comment (fflaten) [reply src/System.Management.Automation/engine/debugger/debugger.cs:985]: Not in this PR. Might be a worth a discussion in a separate issue. There's currently a risk of hitting breakpoints in the wrong file when `file.ps1` and `FILE.ps1` exists in Linux.

This PR simply reverts to original behavior pre 7.4 to fix broken VSCode-extension on Windows. `OrdinalIgnoreCase` is used consistently in the codebase, so a intentional (breaking) change would affect more code, likely outside the debugger as well.

**Review by andyleejordan — APPROVED** [review]
Excellent work @fflaten! I confirmed this resolves the issue. Thanks so much!

File: test/powershell/Language/Scripting/Debugging/DebuggerScriptTests.Tests.ps1
Comment (andyleejordan) [inline test/powershell/Language/Scripting/Debugging/DebuggerScriptTests.Tests.ps1:1]: 🥳

File: src/System.Management.Automation/engine/debugger/debugger.cs (hunk 1 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/debugger/debugger.cs:985]: Please open tracking issue.

Comment (fflaten) [reply src/System.Management.Automation/engine/debugger/debugger.cs:985]: Created #20057 for debugger specifically

**Pull request merged**